[task-5] WIP Создание консольного приложения CalculatorCMD

St1ke opened the pull request.

St1ke committed:
[task-5] WIP Создание консольного приложения CalculatorCMD

Tauders commented on task-5/CalculatorCMD/CalculatorCMD/Program.cs:
```diff
@@ -6,7 +6,46 @@ class Program
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello World!");
+            try
+            {
```
Зачем такой большой try/catch? 

Новое задание: сделай вывод ошибок очень точный, чтобы явно было видно, где ошибся пользователь

Tauders replied:
Новое задание: сделай вывод ошибок очень точный, чтобы явно было видно, где ошибся пользователь

т. е. что бы текст был "Вы ввели первый аргумент неправильно"
"Вы ввели второе число неправильно"

Tauders commented on task-5/CalculatorCMD/CalculatorCMD/Program.cs:
```diff
@@ -6,7 +6,46 @@ class Program
+                if (oper == '+')
+                {
+                    total = a + b;
+                    Console.WriteLine(total);
```
Зачем эта строка повторяется столько раз?

CodingFeles replied:
Замечание так и не было устранено.

Tauders commented on task-5/CalculatorCMD/CalculatorCMD/Program.cs:
```diff
@@ -6,7 +6,46 @@ class Program
+
+                else if (oper == '/')
+                {
+                    total = a / b;
```
Что будет, если b == 0?

St1ke replied:
будет добавление ещё одного блока catch

Tauders commented on task-5/CalculatorCMD/CalculatorCMD/Program.cs:
```diff
@@ -6,7 +6,46 @@ class Program
+            {
+                double a = double.Parse(args[0]);
+                double b = double.Parse(args[1]);
+                char oper = char.Parse(args[2]);
```
Используй нормальные имена, а не сокращения

Tauders commented on task-5/CalculatorCMD/CalculatorCMD/Program.cs:
```diff
@@ -6,7 +6,46 @@ class Program
+                {
+                    total = a - b;
+                    Console.WriteLine(total);
+                }
+
```
Зачем здесь пустая строка?

St1ke committed:
[task-5]Исправление ошибок консольного приложения CalculatorCMD

St1ke committed:
[task-5] Исправление недочётов

1)Сделан точный вывод ошибок
2)Добавлен код для предотвращения деления на 0

Tauders commented on task-5/CalculatorCMD/CalculatorCMD/Program.cs:
```diff
@@ -6,7 +6,57 @@ class Program
+                {
+                    total = number1 / number2;
+                if (number2 == 0)
+                    Console.WriteLine("На ноль делить нельзя!!!");
```
Почему после деления? Попробуй запустить с 0.
Почему нет `{`?

Tauders commented on task-5/CalculatorCMD/CalculatorCMD/Program.cs:
```diff
@@ -6,7 +6,57 @@ class Program
+                Console.WriteLine($"Вы ввели второй аргумент неправильно. Полученная ошибка: {ex.Message}");
+                return;
+            }
+            action = char.Parse(args[2]);
```
А здесь почему нет проверки?

St1ke committed:
[task-5] Завершил создание консольного приложения CalculatorCMD

Исправил недочёты:
1) Сделал вывод ошибок более точечным
2) Исправил ошибку при попытке деления на ноль

CodingFeles reviewed (changes requested):
Хочется чтобы последнее замечание было устранено. После этого работу можно считать завершенной.
